### PR TITLE
Jk scheduler

### PR DESCRIPTION
**Pull request by kuba-krj**
# Description
Implement lr scheduler. Neptune links: [constant scheduler](https://app.neptune.ai/o/pmtest/org/llm-random-tests/runs/details?viewId=standard-view&detailsTab=charts&shortId=LLMTST-226&type=run&lbViewUnpacked=true&sortBy=%5B%22sys%2Fcreation_time%22%5D&sortFieldType=%5B%22datetime%22%5D&sortFieldAggregationMode=%5B%22auto%22%5D&sortDirection=%5B%22descending%22%5D&suggestionsEnabled=false), [cosine scheduler](https://app.neptune.ai/o/pmtest/org/llm-random-tests/runs/details?viewId=standard-view&detailsTab=charts&shortId=LLMTST-230&type=run&lbViewUnpacked=true&sortBy=%5B%22sys%2Fcreation_time%22%5D&sortFieldType=%5B%22datetime%22%5D&sortFieldAggregationMode=%5B%22auto%22%5D&sortDirection=%5B%22descending%22%5D&suggestionsEnabled=false). Image with plot of lr below (respectively: constant with warmup=100; cosine with warmup=100, final_lr_step=900)

# Checklist
   - [x] Is this PR focused on a single feature without the possibility of being divided into smaller PRs?
   - [x] Are all variables set with logical defaults that are backward compatible?
   - [yes] I attached link to neptune if it was necessary
   - [yes] Have you successfully executed the linter and tests?
   - [yes] Are all functions concise and don't require splitting?
   - [yes] Is there documentation for arguments and complex logic?
   - [no] Are there no temporary solutions in the code? Should a task be added to Trello to refine the code?
   - [yes] Are all functions placed in appropriate files?
   - [yes] Is the PR name meaningful and descriptive?
   - [provided] Is PR description provided, or unnecessary?
  
Other questions:
   - [-] If I made changes to the requirements or anything related to the Singularity image I did generate new image and upload a new image to the clusters
   - [yes] If this change is significant enough I notified all individuals working with this repository (@channel in slack channel)
   - [yes] If there is a need for a second reviewer I requested additional review 
 
**Constant scheduler**
image

**Cosine scheduler**
image

### CODE REVIEW

File: full_config0.yaml
Comment (maciek-pioro) [inline full_config0.yaml:1]: Let's remove that

File: lizrd/test.yaml
Comment (maciek-pioro) [inline lizrd/test.yaml:1]: Let's remove this file

File: lizrd/train/scheduler.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        self,
+        lr_warmup_steps: int,
+        lr: float,
+        final_lr_step: int,
+        final_lr_fraction: float,
+    ):
+        self.lr_warmup_steps = lr_warmup_steps
+        self.lr = lr
+        self.final_lr_step = final_lr_step
+        self.final_lr_fraction = final_lr_fraction
```
Comment (maciek-pioro) [inline lizrd/train/scheduler.py:56]: Let's just do asserts on types. (e.g. assert isinstance(final_lr_step, int))

**Review by maciek-pioro — CHANGES_REQUESTED** [review]
Minor changes requested. All in all - well done, much appreciated work.

**Comment by Simontwice** [comment]
I would love to see some simple tests on the schedulers. One for each type.

**Comment by Simontwice** [comment]
That the values each step are correct.

File: lizrd/train/scheduler.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+            lr_warmup_steps=args.lr_warmup_steps,
+            lr=args.learning_rate,
+            final_lr_step=args.final_lr_step,
+            final_lr_fraction=args.final_lr_fraction,
+        )
+    else:
+        raise ValueError(f"Unknown scheduler: {args.scheduler}")
+
+
+class BaseScheduler(ABC):
```
Comment (Simontwice) [inline lizrd/train/scheduler.py:23]: can we rename to `BaseLRScheduler`? methods are already lr specific

**Review by Simontwice — CHANGES_REQUESTED** [review]
naming and tests would be very nice

**Review by Simontwice — APPROVED** [review]

File: lizrd/train/scheduler.py [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+            lr_warmup_steps=args.lr_warmup_steps,
+            lr=args.learning_rate,
+            final_lr_step=args.final_lr_step,
+            final_lr_fraction=args.final_lr_fraction,
+        )
+    else:
+        raise ValueError(f"Unknown scheduler: {args.scheduler}")
+
+
+class BaseLRScheduler(ABC):
```
Comment (maciek-pioro) [inline lizrd/train/scheduler.py:23]: Can we rename this to AbstractLRScheduler? We use this convention all over the place

**Review by maciek-pioro — CHANGES_REQUESTED** [review]
Once again, two smal things

**Review by maciek-pioro — APPROVED** [review]
LGTM, keep up the good work
